.tekton/checks.yaml

```
@@ -83,6 +83,39 @@ spec:
       - name: basic-auth
         workspace: git-auth
 
+    - name: run-go-unit-tests
+      params:
+      - name: SOURCE_ARTIFACT
+        value: $(tasks.clone-repository.results.SOURCE_ARTIFACT)
+      taskSpec:
+        metadata:
+          name: run-go-unit-tests
+          namespace: rh-acs-tenant
+        params:
+        - name: SOURCE_ARTIFACT
+          type: string
+        volumes:
+        - name: workdir
+          emptyDir: { }
+        stepTemplate:
+          volumeMounts:
+          - mountPath: /var/workdir
+            name: workdir
+        steps:
+        - name: use-trusted-artifact
+          image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:9b180776a41d9a22a1c51539f1647c60defbbd55b44bbebdd4130e33512d8b0d
+          args:
+          - use
+          - $(params.SOURCE_ARTIFACT)=/var/workdir/source
+        - name: run-go-tests
+          image: registry.redhat.io/ubi9/go-toolset@sha256:3ce6311380d5180599a3016031a9112542d43715244816d1d0eabc937952667b
+          workingDir: /var/workdir/source
+          script: |
+            #!/usr/bin/env bash
+            set -euo pipefail
+            set -x
+            time make go-test
```

Since `time make go-test` is the only actual command in the step and since here's no use of pipes and environment variables, this can be simplified:
```suggestion
          script: time make go-test
```